doc/adrs/112_add_support_for_multiple_IObjectService.adoc

```diff
@@ -0,0 +1,118 @@
+= ADR-112 - Add support for multiple IObjectService
```

Comment:
- We will definitely move forward with something similar to the second option. There no need for everybody to manage this complexity everywhere.
- I'm not fan of `IObjectServiceProvider` because it does not provide instances of `IObjectService`. I would rather go forward with renaming `ObjectService` to `ComposedObjectService` and introduce an  `IObjectServiceDelegate` which would have to be implemented by those integrating Sirius Components.
- It would be nice to have the default behavior implement some sort of `IDefaultObjectService` (which would be similar to IObjectService). This way, it would allow an application integrating multiple metamodels to define a custom default behavior for every metamodels and small customization where it is necessary in `IObjectServiceDelegate`.
- It should also be noted that the `IObjectService` should/will be separated into smaller interfaces in order to simplify all that (extracting an `ILabelService` and an `IImageService` would be a no brainer). It would also greatly simplify the manipulation of multiple metamodels in a single application and the reusability of metamodels in various Sirius Web applications. No need to copy/paste every time how to retrieve an object from its id if you just want to customize a label.

To sum things up, we need to have an API which would be more similar to the EMF Edit ones without the dependency to EMF.

It should also be noted that this issue was blocking the use of styled strings in the explorer for example.